custom_components/feedparser/sensor.py

```diff
@@ -39,45 +46,81 @@
     vol.Required(CONF_NAME): cv.string,
     vol.Required(CONF_FEED_URL): cv.string,
     vol.Required(CONF_DATE_FORMAT, default='%a, %b %d %I:%M %p'): cv.string,
-    vol.Optional(CONF_SHOW_TOPN, default=9999): cv.positive_int,
+    vol.Optional(CONF_FILTER, default=CONF_FILTER_DEFAULT): vol.All(cv.string, vol.In(CONF_FILTER_OPTIONS)),
+    vol.Optional(CONF_FILTER_VALUE, default=''): cv.template,
     vol.Optional(CONF_INCLUSIONS, default=[]): vol.All(cv.ensure_list, [cv.string]),
     vol.Optional(CONF_EXCLUSIONS, default=[]): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_REMOVE_IMAGE, default=False): cv.boolean,
 })
 
 
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
-    async_add_devices([FeedParserSensor(
-        feed=config[CONF_FEED_URL],
-        name=config[CONF_NAME],
-        date_format=config[CONF_DATE_FORMAT],
-        show_topn=config[CONF_SHOW_TOPN],
-        inclusions=config[CONF_INCLUSIONS],
-        exclusions=config[CONF_EXCLUSIONS]
-        )], True)
+    async_add_devices([FeedParserSensor(hass=hass, config=config)], True)
 
 
 class FeedParserSensor(Entity):
-    def __init__(self, feed: str, name: str, date_format: str, show_topn: str, exclusions: str, inclusions:str):
-        self._feed = feed
-        self._name = name
-        self._date_format = date_format
-        self._show_topn = show_topn
-        self._inclusions = inclusions
-        self._exclusions = exclusions
-        self._state = None
+    def __init__(self, hass, config: dict = None):
+        self.hass = hass
+        self._config = config
+        self._feed = None
+        self._name = None
+        self._date_format = None
+        self._filter = None
+        self._show_topn = 9999
+        self._filter_value = None
+        self._inclusions = None
+        self._exclusions = None
+        self._remove_image = None
+        self._state = 0
         self._entries = []
 
+        if self._config:
+            self._feed = config[CONF_FEED_URL]
+            self._name = config[CONF_NAME]
+            self._date_format = config[CONF_DATE_FORMAT]
+            if CONF_FILTER in self._config:
```

Comment:
I'd rather not have a breaking change. I'm fine with the change overall, but lets compensate for anyone that has topn already set and pass it into the new format.

Reply:
@TJPoorman you have a chance to revisit this?